cmd/nerdctl/build.go

```diff
@@ -28,66 +28,38 @@ import (
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
-	"github.com/urfave/cli/v2"
+	"github.com/spf13/cobra"
 )
 
-var buildCommand = &cli.Command{
-	Name:   "build",
-	Usage:  "Build an image from a Dockerfile. Needs buildkitd to be running.",
-	Action: buildAction,
-	Flags: []cli.Flag{
-		&cli.StringFlag{
-			Name:    "buildkit-host",
-			Usage:   "BuildKit address",
-			EnvVars: []string{"BUILDKIT_HOST"},
-			Value:   defaults.BuildKitHost(),
-		},
-
-		&cli.StringSliceFlag{
-			Name:    "tag",
-			Aliases: []string{"t"},
-			Usage:   "Name and optionally a tag in the 'name:tag' format",
-		},
-		&cli.StringFlag{
-			Name:    "file",
-			Aliases: []string{"f"},
-			Usage:   "Name of the Dockerfile",
-		},
-		&cli.StringFlag{
-			Name:  "target",
-			Usage: "Set the target build stage to build",
-		},
-		&cli.StringSliceFlag{
-			Name:  "build-arg",
-			Usage: "Set build-time variables",
-		},
-		&cli.BoolFlag{
-			Name:  "no-cache",
-			Usage: "Do not use cache when building the image",
-		},
-		&cli.StringFlag{
-			Name:  "progress",
-			Usage: "Set type of progress output (auto, plain, tty). Use plain to show container output",
-			Value: "auto",
-		},
-		&cli.StringSliceFlag{
-			Name:  "secret",
-			Usage: "Secret file to expose to the build: id=mysecret,src=/local/secret",
-		},
-		&cli.StringSliceFlag{
-			Name:  "ssh",
-			Usage: "SSH agent socket or keys to expose to the build (format: default|<id>[=<socket>|<key>[,<key>]])",
-		},
-	},
+func newBuildCommand() *cobra.Command {
+	var buildCommand = &cobra.Command{
+		Use:   "build",
+		Short: "Build an image from a Dockerfile. Needs buildkitd to be running.",
+		RunE:  buildAction,
+	}
+	buildCommand.Flags().String("buildkit-host", defaults.BuildKitHost(), "BuildKit address [$BUILDKIT_HOST]")
+	buildCommand.Flags().StringSliceP("tag", "t", nil, "Name and optionally a tag in the 'name:tag' format")
+	buildCommand.Flags().StringP("file", "f", "", "Name of the Dockerfile")
+	buildCommand.Flags().String("target", "", "Set the target build stage to build")
+	buildCommand.Flags().StringSlice("build-arg", nil, "Set build-time variables")
+	buildCommand.Flags().Bool("no-cache", false, "Name of the Dockerfile")
+	buildCommand.Flags().String("progress", "auto", "Set type of progress output (auto, plain, tty). Use plain to show container output")
+	buildCommand.Flags().StringSlice("secret", nil, "Secret file to expose to the build: id=mysecret,src=/local/secret")
+	buildCommand.Flags().StringSlice("ssh", nil, "SSH agent socket or keys to expose to the build (format: default|<id>[=<socket>|<key>[,<key>]])")
+
+	return buildCommand
 }
 
-func buildAction(clicontext *cli.Context) error {
-	buildkitHost := clicontext.String("buildkit-host")
+func buildAction(cmd *cobra.Command, args []string) error {
+	buildkitHost, err := cmd.Flags().GetString("buildkit-host")
+	if err != nil {
+		return err
+	}
 	if err := buildkitutil.PingBKDaemon(buildkitHost); err != nil {
```

Comment:
`PingBKDaemon` cannot see the envat  `BUILDKIT_HOST`

Reply:
Thanks, fixed.